Sources/Controllers/DashboardOnMap/MapSettings/MapSettingsGpxViewController.swift

```diff
@@ -0,0 +1,592 @@
+//
+//  MapSettingsGpxViewController.swift
+//  OsmAnd Maps
+//
+//  Created by Dmitry Svetlichny on 09.01.2024.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import UIKit
+import UniformTypeIdentifiers
+
+@objc(OAMapSettingsGpxViewController)
+@objcMembers
+final class MapSettingsGpxViewController: OABaseNavbarSubviewViewController {
+    private let previouslyVisibleTracksKey: String = "PreviouslyVisibleGpxFilePaths"
+    private var searchController: UISearchController?
+    private var segmentedControl: UISegmentedControl?
+    private var allGpxList: [OAGPX] = []
+    private var visibleGpxList: [OAGPX] = []
+    private var recentlyVisibleGpxList: [OAGPX] = []
+    private var filteredGpxList: [OAGPX] = []
+    private var selectedGpxTracks: [OAGPX] = []
+    private var previousSelectedSegmentIndex: Int = 0
+    private var isShowingVisibleTracks: Bool = true
+    private var isSearchActive: Bool = false
+    private var isSearchFilteringActive: Bool = false
+    private var isTracksAvailable: Bool = false
+    private var isVisibleTracksAvailable: Bool = false
+    
+    override func commonInit() {
+        loadGpxTracks()
+        loadVisibleTracks()
+        loadRecentlyVisibleTracks()
+    }
+    
+    override func registerNotifications() {
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(_:)), name: UIResponder.keyboardWillShowNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide(_:)), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.tableView.setEditing(true, animated: false)
+        self.tableView.allowsMultipleSelectionDuringEditing = true
+        self.navigationItem.hidesSearchBarWhenScrolling = true
+        self.definesPresentationContext = true
+        updateSelectedRows()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        if let searchController = searchController, searchController.isActive {
+            searchController.isActive = false
+            self.navigationItem.searchController = nil
+        }
+    }
+    
+    override func getTitle() -> String? {
+        localizedString("shared_string_gpx_tracks")
+    }
+    
+    override func getLeftNavbarButtonTitle() -> String? {
+        localizedString("shared_string_cancel")
+    }
+    
+    override func getRightNavbarButtons() -> [UIBarButtonItem]? {
+        guard let searchButton = createRightNavbarButton(nil, iconName: "ic_navbar_search", action: #selector(onSearchButtonClicked), menu: nil),
```

Comment:
mb use from assets? UIImage.icNavbarSearch

Reply:
The createRightNavbarButton function is designed to accept the icon name as a string.